tests/manage/monitoring/conftest.py

```diff
@@ -0,0 +1,134 @@
+        """
+        prometheus = PrometheusAPI()
+        while info.get('run'):
+            alerts_response = prometheus.get(
+                'alerts',
+                payload={
+                    'silenced': False,
+                    'inhibited': False
+                }
+            )
+            assert alerts_response.ok, 'Prometheus API request failed'
+            for alert in alerts_response.json().get('data').get('alerts'):
+                if alert not in alert_list:
+                    logger.info(f"Adding {alert} to alert list")
+                    alert_list.append(alert)
+            time.sleep(10)
+
+    if not measure_after:
+        start_time = time.time()
+
+    # init logging thread that checks for Prometheus alerts
+    # while workload is running
+    # based on https://docs.python.org/3/howto/logging-cookbook.html#logging-from-multiple-threads
+    info = {'run': True}
+    alert_list = []
+
+    logging_thread = threading.Thread(
+        target=prometheus_log,
+        args=(info, alert_list)
+    )
+    logging_thread.start()
+
+    try:
+        result = operation()
+        if measure_after:
+            start_time = time.time()
+        passed_time = time.time() - start_time
+        if minimal_time:
+            additional_time = minimal_time - passed_time
+            if additional_time > 0:
+                time.sleep(additional_time)
+        stop_time = time.time()
+    except KeyboardInterrupt:
+        # Thread should be correctly terminated on next few lines
+        # this is done in case of user interuption to make sure that thread
+        # is terminated correctly
+        pass
+    info['run'] = False
+    logging_thread.join()
+    logger.info(f"Alerts found during measurement: {alert_list}")
+    return {
+        'start': start_time,
+        'stop': stop_time,
+        'result': result,
+        'metadata': metadata,
+        'prometheus_alerts': alert_list
+    }
+
+
+@pytest.fixture(scope="session")
+def workload_stop_ceph_mgr():
+    """
+    Returns:
+        dict: Contains information about `start` and `stop` time for stopping
+            Ceph manager node.
+    """
+    oc = ocp.OCP(
+        kind=constants.DEPLOYMENT,
+        namespace=defaults.ROOK_CLUSTER_NAMESPACE
+    )
+    mgr = 'rook-ceph-mgr-a'
+
+    def stop_mgr():
+        """
+        Downscale Ceph Manager deployment for 6 minutes. First 5 minutes
+        the alert should be in 'Pending'.
+        After 5 minutes it should be 'Firing'.
+        """
+        # run_time of operation
+        run_time = 60 * 6
+        nonlocal oc
+        nonlocal mgr
+        logger.info(f"Downscaling deployment {mgr} to 0")
+        oc.exec_oc_cmd(f"scale --replicas=0 deployment/{mgr}")
+        logger.info(f"Waiting for {run_time} seconds")
+        time.sleep(run_time)
+        return oc.get(mgr)
+
+    measured_op = measure_operation(stop_mgr)
+    logger.info(f"Upscaling deployment {mgr} back to 1")
```

Comment:
I think that upscaling the mgr deployment back to 1 should be part of the test, to see we don't get alert anymore once we have mgr back

Reply:
In test is tested that we don't see the alert anymore. Changing the deployment back in fixture allows me to measure the exact time when the deployment was down and use this fixture in multiple tests.

Reply:
The idea is to separate workload and the test. The workload performs an operation and saves the evidence, while the test performs checks on the evidence, with the performed operation in mind. There may be multiple different tests for the same workload run. For example in this case, I can use this workload fixture in test which will query Prometheus for particular metrics in given timeframe (as returned by the fixture) to check if the values were changed as expected.